src/adapter/src/optimize/materialized_view.rs

```diff
@@ -299,26 +278,45 @@ impl Optimize<GlobalMirPlan<Resolved>> for OptimizeMaterializedView {
         .map_err(OptimizerError::Internal)?;
 
         // Return the plan at the end of this `optimize` step.
-        Ok(GlobalLirPlan { df_desc, df_meta })
+        Ok(GlobalLirPlan {
+            df_desc,
+            df_meta,
+            phantom: PhantomData::<Unresolved>,
+        })
     }
 }
 
-impl GlobalLirPlan {
-    pub fn unapply(self) -> (LirDataflowDescription, DataflowMetainfo) {
-        (self.df_desc, self.df_meta)
-    }
+impl GlobalLirPlan<Unresolved> {
+    /// Produces the [`GlobalLirPlan`] with [`Resolved`] timestamp.
+    pub fn resolve(mut self, as_of: Antichain<Timestamp>) -> GlobalLirPlan<Resolved> {
+        // Set the `as_of` timestamp for the dataflow.
+        self.df_desc.set_as_of(as_of);
 
-    pub fn df_desc(&self) -> &LirDataflowDescription {
-        &self.df_desc
-    }
+        soft_assert_or_log!(
+            self.df_desc.index_exports.is_empty(),
+            "unexpectedly setting until for a DataflowDescription with an index",
+        );
```

Comment:
I think this soft assert can go? It's message is confusing now that we don't attempt to set `until` anymore, and we assert that `index_exports` is empty below anyway.